scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingMethodImpl.java

```diff
@@ -658,14 +658,24 @@ protected boolean createExecution(NodeSet nodeSet, Node node, InternalJob job, I
 
                     tlogger.debug(task.getId(), "deploying");
 
+                    // Dynamically adjust the start-task-timeout according to the number dependency tasks in a merge.
+                    // above 500 parent tasks, it is worth adjusting.
+                    if (taskDescriptor.getParents().size() > 500) {
```

Comment:
does it make sense to make this 500 configurable?

Reply:
I don't think so, honestly

Reply:
The real variables that matters in the formula and that help calculate the start-task-timeout are the number of parents and the PASchedulerProperties.SCHEDULER_STARTTASK_TIMEOUT which is configurable.
This value was defined after running dozens of vars.

Reply:
it would be pointless:
the timeout we calculate depends on the number of parents. The objective is not the find the best timeout value, but to assign a value that guarantees that the task won't fail if its parents results take too long to be fetched.
By letting the user define the SCHEDULER_STARTTASK_TIMEOUT, we only multiply this value every 500 parents.
The 500 value was chosen after dozens of tests, but it doesn't really matter if it's 500 or 1000 since we don't want to tweak that but only to have a big enough timeout value